src/app/pipes/filter-by-keyword.pipe.ts

```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform } from '@angular/core'
+import { ISortOptions, SortType } from '../models/sort-options.model'
+import { IVideoItem } from '../models/video-item.model'
+
+@Pipe({
+  name: 'filterByKeyword',
+})
+export class FilterByKeywordPipe implements PipeTransform {
+  transform(value: IVideoItem[], options: ISortOptions): IVideoItem[] {
+    // console.log(value, options);
+    if (options.sort.type !== SortType.KeyWord) return value
+    return value.filter(el =>
+      el.snippet.title
+        .toLowerCase()
+        .split(' ')
+        .some(word =>
+          options.keywords
+            .trim()
+            .toLowerCase()
+            .split(' ')
+            .some(w => word.includes(w))
```

Comment:
a great time to learn regular expressions here) If you'll have some time left, try to do it with RegExp.

with your approach it shows result even for some weird query like `angular fsdjfij;ffwf fij fei `, I'm not sure if the search should work this way. (at least google doesn't 😄)